### PR TITLE
Additional example application written in Vue.JS

### PR DESCRIPTION
**Pull request by DamianLion**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- Additional example of the angular example written in vue.js

should fullfil the same functionality as the angular example expect the open id integration which will be provided later on.

### CODE REVIEW

File: core/examples/luigi-sample-vue/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Sample luigi application written in Vue.JS
```
Comment (klaudiagrz) [inline core/examples/luigi-sample-vue/README.md]: `Luigi`

Also: this document doesn't have the obligatory Overview section. See the template for the README.md documents:
https://github.com/kyma-project/community/blob/master/guidelines/internal-guidelines/repository-template/template/README.md

File: core/examples/luigi-sample-vue/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Development
+
+This mono repository uses [Lerna](https://lernajs.io/) for project management. Ensure that you have installed lerna, if not then run
```
Comment (klaudiagrz) [inline core/examples/luigi-sample-vue/README.md]: `Ensure that you have installed Lerna. If not, run the following command:`

File: core/examples/luigi-sample-vue/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Development
+
+This mono repository uses [Lerna](https://lernajs.io/) for project management. Ensure that you have installed lerna, if not then run   
+`npm install --global lerna`
+
+To start developing this application, follow these steps:
+
+1. Install dependencies.  
+`lerna bootstrap` executes npm install and links cross-dependencies
```
Comment (klaudiagrz) [inline core/examples/luigi-sample-vue/README.md]: ``The `lerna bootstrap` command executes `npm install` and links cross-dependencies.``

File: core/examples/luigi-sample-vue/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+```bash
+lerna bootstrap
+```
+
+2. Bundle cross-dependencies.  
+`lerna run` runs the given script in every package referenced in lerna.json
```
Comment (klaudiagrz) [inline core/examples/luigi-sample-vue/README.md]: ``The `lerna run` command runs the given script in every package referenced in the `lerna.json` file.``

File: core/examples/luigi-sample-vue/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+```bash
+lerna run bundle
+```
+
+## Run server
```
Comment (klaudiagrz) [inline core/examples/luigi-sample-vue/README.md]: Use H3 for the subsections

``### Run a server``

File: core/examples/luigi-sample-vue/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+### Compiles and hot-reloads for development
```
Comment (klaudiagrz) [inline core/examples/luigi-sample-vue/README.md]: Too many headings. Maybe you could make it a numbered list (1,2,3...) if this is a step-by-step instruction, or simply bullet points?

Also: please make it more descriptive.

File: core/examples/luigi-sample-vue/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+### Run your end-to-end tests
+```
+npm run test:e2e
+```
+
+## Use OpenID Connect
+
+currently only available in the angular sample
```
Comment (klaudiagrz) [inline core/examples/luigi-sample-vue/README.md]: `Using OpenID Connect is available only in the angular sample.`

File: core/examples/luigi-sample-vue/README.md (hunk 1 above)
Comment (maxmarkus) [reply core/examples/luigi-sample-vue/README.md]: I would suggest to pull from `remote:master` again, as we have updated the README.md there which you can use as your starting point.

File: core/examples/luigi-sample-vue/README.md (hunk 7 above)
Comment (DamianLion) [reply core/examples/luigi-sample-vue/README.md]: I have querstion regarding the availability of the OpenID connect functionality. I was not able to find any code in the angular example interacting in any way with authentication. Can I assume this is functionality from luigi which does not depend on implementation in the underlaying application ?

**Comment by DamianLion** [comment]
@dpolitesap @klaudiagrz Thank you for your review!
Before my question regarding gets losts in the outdated section 😉 

I'm still unsure if there is implentation needed in the example for the openid connect functionality. You may can help me out there 😄 https://github.com/kyma-project/luigi/pull/13#discussion_r208858567

**Comment by DamianLion** [comment]
Link to https://github.com/kyma-project/luigi/issues/15

File: core/examples/luigi-sample-vue/README.md (hunk 7 above)
Comment (maxmarkus) [reply core/examples/luigi-sample-vue/README.md]: Authorization is part of core, no extra implementation required on client/sample app side.

File: core/examples/luigi-sample-vue/vue.config.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+const webpack = require('webpack')
```
Comment (maxmarkus) [inline core/examples/luigi-sample-vue/vue.config.js]: This line seems to be obsolete, unused variable.

File: core/examples/luigi-sample-vue/src/views/projects/settings.vue [hunk 9]
```diff
@@ -0,0 +1,71 @@
+<template>
+  <div>
+    <section class="fd-section">
+      <div class="fd-section__header">
+        <h1 class="fd-section__title">Settings of {{ projectId }}</h1>
+      </div>
+      <div class="fd-panel">
+        LuigiClient hasBack/GoBack:<br>
+        {{luigiClient.linkManager().hasBack()}}
+        {{hasBack}}
```
Comment (maxmarkus) [inline core/examples/luigi-sample-vue/src/views/projects/settings.vue]: Are these two lines with `hasBack` required?

File: core/examples/luigi-sample-vue/README.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+2. Install dependencies.
+    ```bash
+    # The `lerna bootstrap` command executes the Node Package Manager (NPM) installation and links cross-dependencies.
+
+    lerna bootstrap
+    ```
+
+3. Bundle the Luigi core.
+    ```bash
+    # Lerna runs the bundle script in every package where the script exists and were referenced in the lerna.json.
```
Comment (klaudiagrz) [inline core/examples/luigi-sample-vue/README.md]: and ~were~ was referenced

**Comment by DamianLion** [comment]
@klaudiagrz I hope my change meets your expectations 😄

**Pull request merged**